compiler/src/dotty/tools/io/AbstractFile.scala

```diff
@@ -232,9 +241,14 @@ abstract class AbstractFile extends Iterable[AbstractFile] {
     val lookup = lookupName(name, isDir)
     if (lookup != null) lookup
     else {
-      val jfile = new JFile(file, name)
-      if (isDir) jfile.mkdirs() else jfile.createNewFile()
-      new PlainFile(jfile)
+      val path = jpath.resolve(name)
+      try {
+        if (isDir) Files.createDirectories(path)
+        else Files.createFile(path)
+      } catch {
+        case _: FileAlreadyExistsException =>
```

Comment:
I guess it should never happen, otherwise `lookupName` would not have returned `null`